### PR TITLE
wql and haml fixes

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/lib/card/query/attributes.rb [hunk 1]
```diff
@@ -4,17 +4,12 @@ class Query
...
-            subquery(
-              c.wql_hash.merge(unjoined: true, context: c.name)
-            )
-          else
+          unless c && c.respond_to?(:wql_hash)
             raise Card::Error::BadQuery,
-                  '"found_by" value must be valid Search, ' \
-                  "but #{c.name} is a #{c.type_name}"
+                              '"found_by" value must be valid Search, ' \
+                              "but #{c.name} is a #{c.type_name}"
```
Comment (ethn) [inline card/lib/card/query/attributes.rb]: Not important, but is there a reason for this spacing?

File: card/lib/card/set/format/haml_views.rb [hunk 2]
```diff
@@ -22,55 +22,65 @@ module Format
...
-        def haml_to_html haml, locals, a_binding=nil
+        def haml_to_html haml, locals, a_binding=nil, debug_info={}
           a_binding ||= binding
           ::Haml::Engine.new(haml).render a_binding, locals || {}
+        rescue Haml::SyntaxError => e
+          info =
+            if debug_info[:path]
+              Pathname.new(debug_info[:path])
+                      .relative_path_from(Pathname.new(Dir.pwd))
+            end
```
Comment (ethn) [inline card/lib/card/set/format/haml_views.rb]: might put this in its own method

File: card/lib/generators/card/template/USAGE [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Description:
+    Generates a haml template.
+
+Example:
+    wagn generate card:template standard type tshirt core
+
+    This will create:
+        mod/standard/template/type/tshirt/core.haml
+
```
Comment (ethn) [inline card/lib/generators/card/template/USAGE]: cool

File: card/mod/core/set/all/templating.rb [hunk 4]
```diff
@@ -21,11 +21,15 @@ def template
...
+def default_type_id default_card=nil
+  default_card ? default_card.type_id : Card.default_type_id
+end
+
 def new_card_template
   default = rule_card :default, skip_modules: true
 
   dup_card = dup
-  dup_card.type_id = default ? default.type_id : Card.default_type_id
+  dup_card.type_id = default_type_id default
```
Comment (ethn) [inline card/mod/core/set/all/templating.rb]: Is there a reason not to put the `rule_card` call in `default_type_id`?

File: card/mod/standard/set/self/search.rb [hunk 5]
```diff
@@ -42,13 +42,18 @@ def extra_paging_path_args
...
+                         "#{keyword}</span>"
     end
     super()
   end
+
+  def search_keyword
+    vars =
+      root.respond_to?(:search_params) ? root.search_params[:vars] : search_params[:vars]
+    vars && vars[:keyword]
+  end
```
Comment (ethn) [inline card/mod/standard/set/self/search.rb]: much more readable, thanks.

File: card/lib/card/query/attributes.rb (hunk 1 above)
Comment (xithan) [inline card/lib/card/query/attributes.rb]: no idea how that happened

File: card/mod/core/set/all/templating.rb (hunk 4 above)
Comment (xithan) [inline card/mod/core/set/all/templating.rb]: it's used a few lines further down

**Pull request merged**